apm-sniffer/apm-sdk-plugin/neo4j-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/neo4j/v4x/TransactionRunInterceptor.java

```diff
@@ -61,13 +61,15 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
     @Override
     public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
             Object ret) throws Throwable {
-        SessionRequiredInfo requiredInfo = (SessionRequiredInfo) objInst.getSkyWalkingDynamicField();
-        if (requiredInfo == null || requiredInfo.getSpan() == null) {
-            return ret;
-        }
+        return ((CompletionStage<?>) ret).thenApply(resultCursor -> {
```

Comment:
@guillaume-alvarez I think we face compiling error for the change. This is no `import`ed? Please make compiling passed locally first.

Reply:
Sorry, I could not push to my branch (git thinks some file names are too long !?), so I have to copy my changes manually from my local repo (where it builds) to the web UI and missed the import...

Reply:
That could be fixed through git setup, I remember this is a kind of issue on Windows?

Reply:
Indeed I'm on Windows, I just checked it out on linux to see if it works more smoothly.

Reply:
It worked much better on Linux!
I cleaned the commits.